src/index/mod.rs

```diff
@@ -26,26 +29,29 @@ struct IndexConfig {
 /// repository has a remote called `origin` and that the branch `master` exists on it.
 ///
 /// [RFC 2141]: https://rust-lang.github.io/rfcs/2141-alternative-registries.html
-fn load_config(repo: &git2::Repository) -> Result<IndexConfig> {
-    let tree = repo
-        .find_commit(repo.refname_to_id("refs/remotes/origin/master")?)?
-        .tree()?;
-    let file = tree
-        .get_name("config.json")
-        .with_context(|| anyhow::anyhow!("registry index missing config"))?;
-    let config = serde_json::from_slice(repo.find_blob(file.id())?.content())?;
+fn load_config(repo: &git::Repository) -> Result<IndexConfig> {
+    let file = repo
+        .rev_parse_single("refs/remotes/origin/master")?
+        .object()?
+        .try_into_commit()?
+        .tree()?
+        .lookup_entry_by_path("config.json")?
+        .with_context(|| anyhow::anyhow!("registry index missing config"))?
+        .oid
```

Comment:
In the next release of `git-repository`, these three lines (`oid.attach(&repo).object()`) become just `object()`.